pages/products/[id].tsx

```diff
@@ -0,0 +1,269 @@
+import Head from 'next/head';
+import Header from '../../components/Header.js';
+import Footer from '../../components/Footer.js';
+import { useState } from 'react';
+import React from 'react';
+import AddToCart from '../../components/AddToCart';
+import Link from 'next/link';
+
+type Item = {
+  id: string;
+  name: string;
+  img: string;
+  amount: number;
+  price: number;
+  info: string;
+};
+
+type Cart = {
+  id: string;
+  name: string;
+  img: string;
+  amount: number;
+  price: number;
+  info: string;
+};
+
+type Products = {
+  id: string;
+  name: string;
+  img: string;
+  amount: number;
+  price: number;
+  info: string;
+};
+type Props = { product: Item };
+
+//write inside props props: Props
+export function Product(props: Props) {
+  if (!props.product) {
+    return <div>Item not found...</div>;
+  }
+  console.log(props.product);
+  const [pieces, setPieces] = useState(1);
+  const [total, setTotal] = useState(props.product.price);
+
+  function handlePieces(e) {
+    setPieces(Number(e.target.value));
+    setTotal(Number(e.target.value) * props.product.price);
+  }
+
+  return (
+    <div>
+      <Head>
+        <link rel="icon" href="/favicon.png" />
+        <link
+          href="https://fonts.googleapis.com/css2?family=DM+Mono&display=swap"
+          rel="stylesheet"
+        ></link>
+        <title>{props.product.name || ''}</title>
+      </Head>
+      <Header />
+      <div className="wrapper">
+        <img src={props.product.img} alt=""></img>
+        <div className="description">
+          <h1>{props.product.name}</h1>
+          <p>{props.product.info}</p>
+          <hr />
+          <p className="info">
+            Dispatched in 8-10 weeks.
+            <br></br>
+            Free shipping within the 3rd district.
+          </p>
+          <p className="price">{props.product.price}€</p>
+
+          <label htmlFor="productNumber">
+            <input
+              data-cy={'input-field/' + props.product.name.toLowerCase()}
+              type="number"
+              min="1"
+              placeholder="Qty"
+              onChange={handlePieces}
```

Comment:
This is missing a connection to the value of the state variable pieces, that's why there's the bug with the quantity not showing up when you first click on it.

See https://github.com/upleveled/react-components-props-state-may-2020/blob/01d3dd6f2e0ea13a78609e67f0f0339088057465/src/Form.js#L23-L24